client/web/src/SourcegraphWebApp.tsx

```diff
@@ -337,6 +341,19 @@ class ColdSourcegraphWebApp extends React.Component<SourcegraphWebAppProps, Sour
             })
         )
 
+        this.subscriptions.add(
+            combineLatest([this.userRepositoriesUpdates, authenticatedUser])
+                .pipe(
+                    switchMap(([, authenticatedUser]) =>
+                        authenticatedUser ? listUserRepositories({ id: authenticatedUser.id, first: 1 }) : of(null)
+                    )
+                )
+                .subscribe(userRepositories => {
+                    const hasUserAddedRepositories = userRepositories !== null && userRepositories.nodes.length > 0
```

Comment:
```suggestion
                    const hasUserAddedRepositories = userRepositories && userRepositories.nodes.length > 0
```

Reply:
Unfortunately I can't use this, because then `hasUserAddedRepositories` has a type of `boolean | null` instead of just boolean 🙁

Reply:
```suggestion
                    const hasUserAddedRepositories = userRepositories?.nodes.length > 0
```

How about optional chaining?

Reply:
(very nit, not blocking, I acknowledge that this is merged, was just passing through 🙂)

Reply:
Good idea, but TS still complains because then we are comparing (number | undefined) with a number (0).